packages/diffhtml/lib/util/parse.js

```diff
@@ -30,234 +23,173 @@ const rawElementsDefaults = [
   'template',
 ];
 
-const selfClosingElementsDefaults = [
-  'source',
-  'embed',
-  'param',
-  'track',
-  'input',
-  'meta',
-  'link',
-  'area',
-  'base',
-  'col',
-  'wbr',
-  'img',
-  'br',
-  'hr',
-];
+// Magic token used for interpolation.
+export const TOKEN = '__DIFFHTML__';
 
-/** @type {any} */
-const kElementsClosedByOpening = {
-  li: { li: true },
-  p: { p: true, div: true },
-  td: { td: true, th: true },
-  th: { td: true, th: true },
-};
+// Used to find where to place supplemental data in the markup.
+const tokenEx = new RegExp(`${TOKEN}([^_]*)__`);
 
-/** @type {any} */
-const kElementsClosedByClosing = {
-  li: { ul: true, ol: true },
-  a: { div: true },
-  b: { div: true },
-  i: { div: true },
-  p: { div: true },
-  td: { tr: true, table: true },
-  th: { tr: true, table: true },
-};
+/**
+ *
+ * @param {VTree} vTree
+ * @param {VTree[]=} vTrees
+ * @return {VTree[]}
+ */
+function flattenFragments(vTree, vTrees = []) {
```

Comment:
This logic along with the dynamic value interpolation is getting migrated to the html tagged template to allow for easier integration with other parsers. The less responsibility this area has, the better.